src/main/perf/StatisticsHelper.java

```diff
@@ -30,6 +31,8 @@
 
 	private final AtomicInteger starts = new AtomicInteger();
 
+	// TODO reconsider "volatile" everywhere here; we don't need synchronization based on how we use this
+
```

Comment:
Seems like we could also make these `final`?

Reply:
A couple booleans but not the pools because they are set in the loop.